src/main/java/net/minecraftforge/client/model/generators/BlockModelBuilder.java

```diff
@@ -18,7 +27,282 @@
+            {
+                return name;
+            }
+
+            public static @Nullable Origin fromString(String originName)
+            {
+                if (Center.getSerializedName().equals(originName))
+                {
+                    return Center;
+                }
+                if (Corner.getSerializedName().equals(originName))
+                {
+                    return Corner;
+                }
+                if (OpposingCorner.getSerializedName().equals(originName))
+                {
+                    return OpposingCorner;
+                }
+                return null;
+            }
+        }
+
+        private Vector3f translation = Vector3f.ZERO;
+        private Quaternion leftRotation = Quaternion.ONE.copy();
+        private Quaternion rightRotation = Quaternion.ONE.copy();
+        private Vector3f scale = ONE;
+
+        private @Nullable Origin origin;
+        private @Nullable Vector3f originVec;
+
+        private RootTransformBuilder()
+        {
+
+        }
+
+        /**
+         * Sets the translation of the root transform.
+         *
+         * @param translation the translation
+         * @return this builder
+         * @throws NullPointerException if {@code translation} is {@code null}
+         */
+        public RootTransformBuilder translation(Vector3f translation)
+        {
+            this.translation = Preconditions.checkNotNull(translation, "Translation must not be null");
+            return this;
+        }
+
+        /**
+         * Sets the left rotation of the root transform.
+         *
+         * @param rotation the left rotation
+         * @return this builder
+         * @throws NullPointerException if {@code rotation} is {@code null}
+         */
+        public RootTransformBuilder rotation(Quaternion rotation)
+        {
+            this.leftRotation = Preconditions.checkNotNull(rotation, "Rotation must not be null");
+            return this;
+        }
+
+        /**
+         * Sets the left rotation of the root transform.
+         *
+         * @param leftRotation the left rotation
+         * @return this builder
+         * @throws NullPointerException if {@code leftRotation} is {@code null}
+         */
+        public RootTransformBuilder leftRotation(Quaternion leftRotation)
+        {
+            return rotation(leftRotation);
+        }
+
+        /**
+         * Sets the right rotation of the root transform.
+         *
+         * @param rightRotation the right rotation
+         * @return this builder
+         * @throws NullPointerException if {@code rightRotation} is {@code null}
+         */
+        public RootTransformBuilder rightRotation(Quaternion rightRotation)
+        {
+            this.rightRotation = Preconditions.checkNotNull(rightRotation, "Rotation must not be null");
+            return this;
+        }
+
+        public RootTransformBuilder postRotation(Quaternion postRotation) {
```

Comment:
Formatting: braces. Also, this should have javadocs.
```suggestion
        public RootTransformBuilder postRotation(Quaternion postRotation)
        {
```